lib/expressions.js

```diff
@@ -29,7 +29,10 @@ function primaryName(expression) {
   if (expression.name) {
     expression = expression.name;
   }
-  if (expression.indexOf('\'') === 0 || expression.match(/^[0-9]/)) {
+  if (expression.match(/^('|").*('|")$/)) { // string literal
+    return '';
+  }
+  if (expression.match(/^-?\d+/)) { // number literal
```

Comment:
This regexp will match any expression that starts with a number like `123abc`. Consider this one  instead, which ensures that there's at most one decimal point, and that the expression is entirely numeric.

`/^-?\d+(\.\d+)?$/`